core/src/main/scala/com/eharmony/spotz/backend/ParallelFunctions.scala

```diff
@@ -0,0 +1,82 @@
+package com.eharmony.spotz.backend
+
+import com.eharmony.spotz.objective.Objective
+import com.eharmony.spotz.optimizer.grid.Grid
+import com.eharmony.spotz.optimizer.hyperparam.RandomSampler
+
+import scala.reflect.ClassTag
+
+/**
+  * Use scala parallel collections to parallelize the optimizer computation.
+  */
+trait ParallelFunctions extends BackendFunctions {
```

Comment:
I had considered this.  The amount of code duplication is small enough.. ~ 3 lines, that I didn't feel it was absolutely  necessary.  When more optimizer algorithms are added and assuming there's still code duplication, I'll refactor it.